src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strJoin = '';
+
+    if (this.length === 0) {
+      return strJoin;
+    }
+
+    for (let value of this) {
+      if (this.length === 1) {
+        strJoin += value;
+
+        return strJoin;
+      }
+
+      if (value === undefined || value === null) {
```

Comment:
```suggestion
      if (value !== undefined || value !== null) {
```
Let's refactor this code a bit and make it more simple and short. you are repeating 
strJoin += `${separator}${value}`; two times and it's not good.
You can check if the element (`value`) is not null or undefined, and if so - add the value to the string (`strJoin += value`). If it's null or undefined - do nothing.
Also, you need to add a separator (`strJoin += separator`) to your string if it's not the last element of the array